src/Configuration/Properties/PropertyCollection.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Integration\Configuration\Properties;
+
+use Illuminate\Support\Arr;
+use MyParcelCom\Integration\Collection;
+
+/**
+ * @extends Collection<array-key, Property>
+ */
+class PropertyCollection extends Collection
```

Comment:
My previous comment about the separation of two concepts also impacts this class. Because after the old _Property_ class becomes _Field_ then this collection ain't a _PropertyCollection_ anymore. 

As soon as the _Property_ class gets changed into a _Field_ class, and a json-schema tight _Property_ class appears, this class would have a wrong name.

We could call it just a `Form`, regardless that it extends a collection. 

It might bother you that it extends a collection but the name is not reflecting that. That came to my mind too, which made me question if extending the laravel collection is really necessary. What does the Laravel collection really bring here? Imagine that you have a constructor that accepts array of fields, or `...` of fields for enforced typing. Then the subsequent operations on that array of fields are simple with native php array functions, such as array_map for example.

Reply:
it's not extending Laravel's Collection class, but our own `MyParcelCom\Integration\Collection`, which is a simple wrapper for an ArrayObject class with a default `toArray` function defined in it

Reply:
ah alright, still the same argument applies.